lib/composer.js

```diff
@@ -194,6 +194,10 @@ export default class Composer extends Disposable {
 
     await Promise.all(jobs)
 
+    if (this.shouldCleanAfterBuild()) {
```

Comment:
Starting at line 195, I suggest:

```javascript
const statuses = await Promise.all(jobs)

if (statuses.every(status => status) && this.shouldCleanAfterBuild()) {
  this.clean(false)
}
```

Change the signature of `clean` to `async clean (startGroup = true)` and put a guard conditional before the log `group`/`groupEnd` calls. For instance,

```javascript
if (startGroup) latex.log.group('LaTeX Clean')
```